trade-finance-manager-api/src/v1/controllers/user/helpers/loginCallback.helper.test.js

```diff
@@ -0,0 +1,199 @@
+
+    expect(result).toEqual({ user: USER, tokenObject: TOKEN_OBJECT });
+  });
+
+  it("returns a 'userOrPasswordIncorrect' error when the user doesn't exist", async () => {
+    mockFindByUsernameReturnsNullUser();
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(USER);
+    mockUpdateLastLoginSuccess(USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userOrPasswordIncorrect });
+  });
+
+  it('returns an error if findByUsername returns an error message', async () => {
+    mockFindByUsernameReturnsError();
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(USER);
+    mockUpdateLastLoginSuccess(USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: ERROR });
+  });
+
+  it("returns a 'userOrPasswordIncorrect' error when the password is incorrect", async () => {
+    mockFindByUsernameSuccess(USER);
+    mockValidPasswordFailure();
+    mockIssueJWTSuccess(USER);
+    mockUpdateLastLoginSuccess(USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userOrPasswordIncorrect });
+  });
+
+  it("returns a 'userIsDisabled' error when the user is disabled", async () => {
+    const DISABLED_USER = { ...USER, disabled: true };
+
+    mockFindByUsernameSuccess(DISABLED_USER);
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(DISABLED_USER);
+    mockUpdateLastLoginSuccess(DISABLED_USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userIsDisabled });
+  });
+
+  it("returns a 'userIsBlocked' error when the user is blocked", async () => {
+    const BLOCKED_USER = { ...USER, status: 'blocked' };
+
+    mockFindByUsernameSuccess(BLOCKED_USER);
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(BLOCKED_USER);
+    mockUpdateLastLoginSuccess(BLOCKED_USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userIsBlocked });
+  });
+
+  it("returns a 'userOrPasswordIncorrect' error when the user doesn't exist and the user is disabled", async () => {
+    const DISABLED_USER = { ...USER, disabled: true };
+
+    mockFindByUsernameReturnsNullUser();
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(DISABLED_USER);
+    mockUpdateLastLoginSuccess(DISABLED_USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userOrPasswordIncorrect });
+  });
+
+  it("returns a 'userOrPasswordIncorrect' error when the user doesn't exist and the user is blocked", async () => {
+    const BLOCKED_USER = { ...USER, status: 'blocked' };
+
+    mockFindByUsernameReturnsNullUser();
+    mockValidPasswordSuccess();
+    mockIssueJWTSuccess(BLOCKED_USER);
+    mockUpdateLastLoginSuccess(BLOCKED_USER);
+
+    const result = await login(USERNAME, PASSWORD);
+
+    expect(result).toEqual({ error: userOrPasswordIncorrect });
+  });
```

Comment:
I'm slightly unsure about these 2, e.g.
> when the user doesn't exist and the user is blocked

How can the user be blocked if they don't exist?

Reply:
Great point --removed